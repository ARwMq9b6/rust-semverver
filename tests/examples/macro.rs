#![feature(decl_macro, use_extern_macros)]

pub mod old {
    pub struct Item;

    pub macro bar() { Item }

    fn abc() -> Item {
        bar!()
    }
}

pub mod new {
    pub struct Item;

    #[macro_export]
    macro_rules! bar {
        () => {
            Item
        }
    }

    #[allow(dead_code)]
    fn abc() -> Item {
        bar!()
    }
}
