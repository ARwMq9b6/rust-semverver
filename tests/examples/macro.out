version bump: 1.0.0 -> (Breaking) -> 2.0.0
warning: Removal of at least one path to `bar` (Breaking)
 --> $REPO_PATH/tests/examples/macro.rs:6:9
  |
6 |     pub macro bar() { Item }
  |         ^^^^^^^^^^^^^^^^^^^^

